### ISSUE TITLE
[HF21 Testnet] Unexpected date overflow

### ISSUE THREAD

**Issue by Gandalf-the-Grey** [steemit/steem#3403]
While creating a proposal with end date "far in the future" (useful for refund proposals) I get unexpected overflow, where desired year `2050` became `1913`

`create_proposal gtg steem.dao "2019-07-01T00:00:00" "2050-01-01T00:00:00" "0.001 TBD" "Issue: Unexpected date overflow #3403" gtg-sps-3 true`

results in:
```
{
  "operations": [[
      "create_proposal",{
        "creator": "gtg",
        "receiver": "steem.dao",
        "start_date": "2019-07-01T00:00:00",
        "end_date": "1913-11-25T17:31:44",
        "daily_pay": "0.001 TBD",
        "subject": "Issue: Unexpected date overflow #3403",
        "permlink": "gtg-sps-3"
      }
    ]
  ],
```

**Comment by joticajulian** [comment]
The date itself is ok because `time_point_sec` uses unsigned int32, and there should no be problems processing the proposals.

**The problem is in the conversion from time to string. That is, only the visualization.** Look that it uses `int32_t` but not unsigned https://github.com/steemit/steem/blob/a470bc283260e582b56897bd5c68107aa0289f18/libraries/fc/src/time.cpp#L25-L29

Maybe changing it to `uint32_t` or `int64_t` solves the problem.

This is related to the [Year 2038 problem](https://en.wikipedia.org/wiki/Year_2038_problem).

**Comment by sgerbino** [comment]
Yes this is true. But there is more to the problem. Between upgrading boost versions they went from handling time with `int32_t` to `int64_t`. Operation that stored special dates with `-1` would not come out correct if we just changed it.

**Comment by mvandeberg** [comment]
Should be fixed in a8b3ddbc3

@sgerbino to close
